services/QuillLMS/app/services/clever_integration/classroom_creator.rb

```diff
@@ -28,12 +28,33 @@ def run
       )
     end
 
+    private def name
+      data[:name].present? ? valid_name : "Classroom #{clever_id}"
+    end
+
+    private def other_owned_classroom_names
+      @other_owned_classroom_names ||= teacher.classrooms_i_own.pluck(:name)
+    end
+
     private def role
       OWNER
     end
 
     private def synced_name
-      name
+      data[:name]
+    end
+
+    private def teacher
+      ::User.find(teacher_id)
+    end
+
+    private def valid_name
+      temp_name = data[:name]
+
+      loop do
+        return temp_name unless other_owned_classroom_names.include?(temp_name)
+        temp_name += '_1'
```

Comment:
That was the intention, but I agree that `_1`, `_2` makes more sense.  I've extracted out a new [method](https://github.com/empirical-org/Empirical-Core/pull/8653/commits/142911eb16a5df27a6685e322a912f53046f8c96) to handle this since `{Google,Clever}Integration::Classroom{Creator,Updater}` were all using this same pattern.